VoodooI2CHID/VoodooI2CTouchscreenHIDEventDriver.cpp

```diff
@@ -25,67 +25,52 @@ OSDefineMetaClassAndStructors(VoodooI2CTouchscreenHIDEventDriver, VoodooI2CMulti
 // Override of VoodooI2CMultitouchHIDEventDriver
 
 bool VoodooI2CTouchscreenHIDEventDriver::checkFingerTouch(AbsoluteTime timestamp, VoodooI2CMultitouchEvent event) {
-    bool got_transducer = false;
+    absolutetime_to_nanoseconds(timestamp, &last_interaction);
     
-    // If there is a finger touch event, decide if it is single or multitouch.
+    // Avoid phantom clicks happening after scrolling
+    if ((last_interaction - last_multitouch_interaction) < MULTITOUCH_TIMEOUT) {
+        return false;
+    }
+
+    VoodooI2CDigitiserTransducer* transducer = OSDynamicCast(VoodooI2CDigitiserTransducer, event.transducers->getObject(0));
     
-    for (int index = 0; index < digitiser.contact_count->getValue() + 1; index++) {
```

Comment:
Does anyone know why there was a `for` loop here when this function is only ever called for `event.contact_count < 1`?

Also, why did it go to `digitiser.contact_count->getValue() + 1`? It seems like that would always be out of bounds.

Reply:
Here seems to be the culprit. Finger is not always the first transducer. It seems that pen is the first one on my WCOM48CE.